### PR TITLE
Adding support for sending user input for the ballerina program

### PR DESCRIPTION
**Pull request by rasika**

### CODE REVIEW

File: composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java [hunk 1]
```diff
@@ -146,91 +145,116 @@ public void run() {
...
+                        // This is to handle the cloud use case.
+                        if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD)
+                                && getServerStartedURL() != null) {
+                            this.updatePort(getServerStartedURL());
+                            line = LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD + " "
+                                    + getServerStartedURL();
+                        }
+
+                        // This is to handle local service run use case.
+                        if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_LOCAL)
```
Comment (nadeeshaan) [inline composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java]: Shall we simplify the If-Else Clause by taking out the pushToClient method call out?

File: composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java [hunk 2]
```diff
@@ -146,91 +145,116 @@ public void run() {
...
             }
         } catch (IOException e) {
             logger.error("Error while sending error stream to client.", e);
-        } finally {
-            if (reader != null) {
-                IOUtils.closeQuietly(reader);
+        }
+    }
+
+    public void pushMessageToProcess(String[] input) {
```
Comment (nadeeshaan) [inline composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java]: Let's add a doc comment for the public method

File: composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java (hunk 1 above)
Comment (rasika) [reply composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java]: I was reluctant to change the existing logic as the second `If` logic has a dependency on the first `If` logic(aka. `line` variable)

File: composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java (hunk 2 above)
Comment (rasika) [reply composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java]: Let's make it `private` method

**Pull request merged**